ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -459,6 +469,84 @@ public void handleDeleteRequest(Request request) throws IOException, Interrupted
             metrics.deleteBlobTotalTimeInMs, null, null, totalTimeSpent));
   }
 
+  @Override
+  public void handleTtlUpdateRequest(Request request) throws IOException, InterruptedException {
+    TtlUpdateRequest updateRequest =
+        TtlUpdateRequest.readFrom(new DataInputStream(request.getInputStream()), clusterMap);
+    long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+    long totalTimeSpent = requestQueueTime;
+    metrics.updateBlobTtlRequestQueueTimeInMs.update(requestQueueTime);
+    metrics.updateBlobTtlRequestRate.mark();
+    long startTime = SystemTime.getInstance().milliseconds();
+    TtlUpdateResponse response = null;
+    try {
+      ServerErrorCode error =
+          validateRequest(updateRequest.getBlobId().getPartition(), RequestOrResponseType.TtlUpdateRequest, false);
+      if (error != ServerErrorCode.No_Error) {
+        logger.error("Validating TtlUpdateRequest failed with error {} for request {}", error, updateRequest);
+        response = new TtlUpdateResponse(updateRequest.getCorrelationId(), updateRequest.getClientId(), error);
+      } else {
+        BlobId convertedStoreKey =
+            (BlobId) getConvertedStoreKeys(Collections.singletonList(updateRequest.getBlobId())).get(0);
+        MessageFormatInputStream stream =
+            new TtlUpdateMessageFormatInputStream(convertedStoreKey, convertedStoreKey.getAccountId(),
+                convertedStoreKey.getContainerId(), updateRequest.getExpiresAtMs(), updateRequest.getOperationTimeInMs());
```

Comment:
Can we deny the request if expireTime is not infinite? 
Raise storeException `Update_Not_Allowed` may be a choice.

Reply:
I can do that - will add that piece of code in the store.

Reply:
Done